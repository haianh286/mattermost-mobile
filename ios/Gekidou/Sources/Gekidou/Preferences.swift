//
//  File.swift
//  
//
//  Created by Elias Nahum on 26-06-22.
//

import Foundation

let appGroupId = Bundle.main.infoDictionary!["AppGroupIdentifier"] as? String

public class Preferences: NSObject {
    private let userDefauls = UserDefaults(suiteName: appGroupId)
    
    @objc public static let `default` = Preferences()
    
    override private init() {}
    
    public func object(forKey name: String) -> Any? {
        return userDefauls?.object(forKey: name)
    }
    
    public func removeObject(forKey name: String) {
        userDefauls?.removeObject(forKey: name)
    }
    
    public func set(_ value: Any?, forKey key: String) {
        userDefauls?.set(value, forKey: key)
    }
}
