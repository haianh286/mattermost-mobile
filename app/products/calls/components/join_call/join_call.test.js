// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';
import {Alert} from 'react-native';

import Preferences from '@mm-redux/constants/preferences';

import JoinCall from './join_call';

describe('JoinCall', () => {
    const baseProps = {
        actions: {
            joinCall: jest.fn(),
        },
        theme: Preferences.THEMES.denim,
        call: {
            participants: {
                'user-1-id': {
                    id: 'user-1-id',
                    muted: false,
                    isTalking: false,
                },
                'user-2-id': {
                    id: 'user-2-id',
                    muted: true,
                    isTalking: true,
                },
            },
            channelId: 'channel-id',
            startTime: 100,
            speakers: 'user-2-id',
            screenOn: false,
            threadId: false,
        },
        confirmToJoin: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<JoinCall {...baseProps}/>);

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should join on click', () => {
        const joinCall = jest.fn();
        const props = {...baseProps, actions: {joinCall}};
        const wrapper = shallow(<JoinCall {...props}/>);

        wrapper.simulate('press');
        expect(Alert.alert).not.toHaveBeenCalled();
        expect(props.actions.joinCall).toHaveBeenCalled();
    });

    test('should ask for confirmation on click', () => {
        const joinCall = jest.fn();
        const props = {...baseProps, confirmToJoin: true, actions: {joinCall}};
        const wrapper = shallow(<JoinCall {...props}/>);

        wrapper.simulate('press');
        expect(Alert.alert).toHaveBeenCalled();
        expect(props.actions.joinCall).not.toHaveBeenCalled();
    });
});
