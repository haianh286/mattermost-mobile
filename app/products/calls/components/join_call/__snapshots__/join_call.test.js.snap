// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinCall should match snapshot 1`] = `
<Pressable
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3DB887",
      "flexDirection": "row",
      "height": 38,
      "justifyContent": "center",
      "padding": 5,
      "width": "100%",
    }
  }
>
  <CompassIcon
    name="phone-in-talk"
    size={16}
    style={
      Object {
        "color": "#ffffff",
        "marginLeft": 10,
        "marginRight": 5,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Join Call
  </Text>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "flex": 1,
        "fontWeight": "400",
        "marginLeft": 10,
      }
    }
  >
    <FormattedRelativeTime
      updateIntervalInSeconds={1}
      value={100}
    />
  </Text>
  <View
    style={
      Object {
        "marginRight": 5,
      }
    }
  >
    <Connect(Avatars)
      breakAt={1}
      listTitle={
        <InjectIntl(FormattedText)
          defaultMessage="CALL PARTICIPANTS"
          id="calls.join_call.participants_list_header"
          style={
            Object {
              "color": "rgba(63,67,80,0.56)",
              "fontSize": 12,
              "fontWeight": "600",
              "paddingHorizontal": 16,
              "paddingVertical": 0,
              "top": 16,
            }
          }
        />
      }
      userIds={
        Array [
          "user-1-id",
          "user-2-id",
        ]
      }
    />
  </View>
</Pressable>
`;
